src/components/clubdetail/CurrentlyReadingCard/CurrentlyReadingCard.tsx

```diff
@@ -1,15 +1,135 @@
+import { useEffect, useState } from 'react';
+import { Link, useParams } from 'react-router-dom';
+
 import { ClubDetailCardWrapper } from '../ClubDetailCardWrapper/ClubDetailCardWrapper';
+import { BookType } from '../../../model/Book';
+import {
+  getClubsCurrentBook,
+  removeClubsCurrentBook,
+} from '../../../api/clubsCurrentlyReading';
+import { fetchBookDetails } from '../../../api/bookDetailApi';
+import { Button } from '../../Button/Button';
+import { ConfirmDialog } from '../../ConfirmDialog/ConfirmDialog';
+import { Loading } from '../../Loading/Loading';
+import style from './CurrentlyReadingCard.module.scss';
 
 type CurrentlyReadingCardProps = {
+  clubId: number;
   title: string;
-  text: string;
+  isOwner: boolean;
 };
 
 export const CurrentlyReadingCard: React.FC<CurrentlyReadingCardProps> = ({
   title,
-  text,
-}) => (
-  <ClubDetailCardWrapper title={title} text={text}>
-    <p>lalala</p>
-  </ClubDetailCardWrapper>
-);
+  clubId,
+  isOwner,
+}) => {
+  const { id } = useParams<{ id: string }>();
+
+  const [currentBooks, setCurrentBooks] = useState<BookType[]>([]);
+  const [bookToRemove, setBookToRemove] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchCurrentBooks = async () => {
+      setIsLoading(true);
+
+      try {
+        const booksData = await getClubsCurrentBook(clubId);
+
+        if (booksData?.length) {
+          const bookPromises = booksData.map((bookData) =>
+            fetchBookDetails({
+              editionId: bookData.currentBookId,
+              authorKey: bookData.authorKey,
+            }),
+          );
+
+          const books = await Promise.all(bookPromises);
+          setCurrentBooks(books.filter((book) => book !== null));
+          setIsLoading(false);
+        }
+      } catch (error) {
+        console.error('Error fetching current books:', error);
+      }
+    };
+
+    fetchCurrentBooks();
+  }, [clubId]);
+
+  const handleRemoveClick = (bookId: string) => {
+    setBookToRemove(bookId);
+  };
+
+  const handleConfirmDelete = async () => {
+    if (bookToRemove) {
+      try {
```

Comment:
Same here. `removeClubsCurrentBook` will never throw because if it does, you are handling it with the try-catch inside of the functuion